### PR TITLE
Fix some bad constants and incorrectly named files found by zeitwerk

### PR DESCRIPTION
**Pull request by jrafanie**
* The constant was seemingly subclassing itself???  Maybe I'm wrong but the constant couldn't be found.
* Renamed plural model filename with singular class to singular filename.

### CODE REVIEW

File: app/models/manageiq/providers/cisco_intersight/physical_infra_manager/physical_switch_network_port.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module ManageIQ::Providers::CiscoIntersight
-  class PhysicalInfraManager::PhysicalSwitchNetworkPort < ::PhysicalSwitchNetworkPort
+  class PhysicalInfraManager::PhysicalSwitchNetworkPort < ::PhysicalNetworkPort
```
Comment (agrare) [inline app/models/manageiq/providers/cisco_intersight/physical_infra_manager/physical_switch_network_port.rb:2]: :+1: this is correct, I'm not sure this needs to be `PhysicalSwitchNetworkPort` and not `PhysicalNetworkPort` but can handle that separately.

Comment (jrafanie) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/physical_switch_network_port.rb:2]: Honestly, I thought it was autocomplete that caused this and the next one. I did stare at the error a few times.  🤣

File: app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module ManageIQ::Providers::CiscoIntersight
-  class PhysicalInfraManager::Vm < ManageIQ::Providers::PhysicalInfraManager::Vm
+  class PhysicalInfraManager::Vm < ManageIQ::Providers::InfraManager::Vm
```
Comment (agrare) [inline app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb:2]: `PhysicalInfra` is different from `InfraManager` I'm not sure we want to simply subclass `InfraManager::Vm`.  Ideally we would create a `ManageIQ::Providers::PhysicalInfraManager::Vm < ::Vm` subclass in core and subclass that here.

File: app/models/manageiq/providers/cisco_intersight/physical_infra_manager/physical_switch_network_port.rb (hunk 1 above)
Comment (agrare) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/physical_switch_network_port.rb:2]: Yeah they intended to call it this, here is the inventory collection showing the `physical_switch_network_ports` name and the `PhysicalNetworkPort` base class https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/inventory/persister/builder/physical_infra_manager.rb#L86-L92

File: app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb (hunk 2 above)
Comment (jrafanie) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb:2]: Makes sense to me.  Should I drop this part for now?  I can keep this local until it's fixed while I do more zeitwerk work.  Let me know.

Comment (agrare) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb:2]: Yes I'll throw a core PR up quickly so this will be a valid class

Comment (agrare) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb:2]: https://github.com/ManageIQ/manageiq/pull/22473

Comment (jrafanie) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb:2]: testing it now...

Comment (jrafanie) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb]: works now... the constant is defined 👍

Comment (jrafanie) [reply app/models/manageiq/providers/cisco_intersight/physical_infra_manager/vm.rb]: I've removed this change.

**Comment by jrafanie** [comment]
Updated after testing verifying `ManageIQ::Providers::PhysicalInfraManager::Vm` constant is now found.

**Review by agrare — APPROVED** [review]

**Pull request merged**